sdks/python/apache_beam/transforms/create_test.py

```diff
@@ -0,0 +1,123 @@
+
+  def check_read(self, values, coder):
+    source = Create._create_source(values, coder)
+    read_values = source_test_utils.readFromSource(source)
+    self.assertEqual(sorted(values), sorted(read_values))
+
+  def test_create_source_read_with_initial_splits(self):
+    self.check_read_with_initial_splits([], self.coder, num_splits=2)
+    self.check_read_with_initial_splits([1], self.coder, num_splits=2)
+    values = range(8)
+    # multiple values with a single split.
+    self.check_read_with_initial_splits(values, self.coder, num_splits=1)
+    # multiple values with a single split with a large desired bundle size
+    self.check_read_with_initial_splits(values, self.coder, num_splits=0.5)
+    # multiple values with many splits.
+    self.check_read_with_initial_splits(values, self.coder, num_splits=3)
+    # multiple values with uneven sized splits.
+    self.check_read_with_initial_splits(values, self.coder, num_splits=4)
+    # multiple values with num splits equal to num values.
+    self.check_read_with_initial_splits(values, self.coder,
+                                        num_splits=len(values))
+    # multiple values with num splits greater than to num values.
+    self.check_read_with_initial_splits(values, self.coder, num_splits=30)
+
+  def check_read_with_initial_splits(self, values, coder, num_splits):
+    """A test that splits the given source into `num_splits` and verifies that
+    the data read from original source is equal to the union of the data read
+    from the split sources.
+    """
+    source = Create._create_source(values, coder)
+    desired_bundle_size = source._total_size / num_splits
+    splits = source.split(desired_bundle_size)
+    splits_info = [
+        (split.source, split.start_position, split.stop_position)
+        for split in splits]
+    source_test_utils.assertSourcesEqualReferenceSource((source, None, None),
+                                                        splits_info)
+
+  def test_create_source_read_reentrant(self):
+    source = Create._create_source(range(9), self.coder)
+    source_test_utils.assertReentrantReadsSucceed((source, None, None))
+
+  def test_create_source_read_reentrant_with_initial_splits(self):
+    source = Create._create_source(range(24), self.coder)
+    for split in source.split(desired_bundle_size=5):
+      source_test_utils.assertReentrantReadsSucceed((split.source,
+                                                     split.start_position,
+                                                     split.stop_position))
+
+  def test_create_source_dynamic_splitting(self):
+    # 2 values
+    source = Create._create_source(range(2), self.coder)
+    source_test_utils.assertSplitAtFractionExhaustive(source)
+    # Multiple values.
+    source = Create._create_source(range(11), self.coder)
+    source_test_utils.assertSplitAtFractionExhaustive(
+        source, perform_multi_threaded_test=True)
+    source_test_utils.assertSplitAtFractionExhaustive(
```

Comment:
I don't think you need this since the run with 'perform_multi_threaded_test=True' fully covers this assertion.